### DISCUSSION TITLE
procedure for migrating tap formula to pypi package ?

### DISCUSSION THREAD

**Discussion by ipatch** [Homebrew/discussions#3300] [Tap maintenance and Homebrew development] [answered]
hello 👋,

is there a graceful way to migrate a formula to pypi package. as one of the maintainers for the homebrew-freecad tap we currently have a matplotlib formula checked in our tap, but i'm hoping that i can sunset that formula and redirect installs to use the pypi package?

i know homebrew supports the renaming of formula and also the migration of formula from one tap to another but didn't see any documentation of migrating to different packaging platforms 🧐

and if homebrew current does not support such feature, what would be the best course of action? hacky solutions welcomed.

**Comment by jonchang** [comment, marked as answer]
You can create a message that will be shown to users using the `deprecate!` and `disable!` commands suggesting that they migrate to a different packaging system. See our [API docs](https://rubydoc.brew.sh/DeprecateDisable.html) for details and you can grep the homebrew/core repository for examples of how to use it. Also take a look at this documentation page: https://docs.brew.sh/Deprecating-Disabling-and-Removing-Formulae
